### ISSUE TITLE
Using pandas naive `.groupby()` internally should inherit `dropna` and `observed`

### ISSUE THREAD

**Issue by pwwang** [pwwang/datar#138]
Across the package. Probably `sort` as well?

See also #137

**Comment by ftobin** [comment]
Oh wow, I didn't realize that dropna was a thing for #137.  I expected that np.nans would be a normal grouping variable, like in dplyr  :-)

```
> mtcars %>% mutate(cyl = if_else(cyl == 6, NaN, cyl)) %>% group_by(cyl) %>% summarize(max(mpg))
# A tibble: 3 × 2
    cyl `max(mpg)`
  <dbl>      <dbl>
1     4       33.9
2     8       19.2
3   NaN       21.4
```

I think I it kind of depends if you want the dplyr-ish version (defaults to not dropping NaNs) or pandas default (which I highly dislike :-) )

**Comment by pwwang** [comment]
The philosophy of `datar` is trying to align APIs with `dplyr`/`tidyverse`'s APIs. So by default, `datar.dplyr.group_by()` keeps the NA groups. The groups in `datar` are still maintained by `pandas` internally. However, I just missed in some places to pass the arguments to pandas `groupby`, which defaults to drop the NA groups. 

This should be fixed by commit: 8620dc61bf28246ac78f5e9febffdd04ec6d3220
